File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.java
/*
    연속된 문자 처리 -> 스택
*/

import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        
        for(int i=0;i<s.length();i++){
            if(!stack.isEmpty() && stack.peek() == s.charAt(i)){
                stack.pop();
            }else{
                stack.add(s.charAt(i));
            }
        }
        
        if(stack.isEmpty()){
            return 1;
        }else{
            return 0;
        }
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    
    list = []
    
    for i in s:
        if len(list) == 0:
            list.append(i)
        elif list[len(list)-1] == i:
            list.pop()
        else:
            list.append(i)
    
    if len(list) == 0:
        answer = 1
    
    return answer